<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  
  
  <title>Explanation on DCA (Deep Count Autoencoder) | Something about Learning</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <meta name="description" content="Recently, machine learning techniques become more and more popular in the field of bioinformatics, especially for analyzing DNA or RNA sequences. The ability of obtaining single-cell RNA sequence (scR">
<meta property="og:type" content="article">
<meta property="og:title" content="Explanation on DCA (Deep Count Autoencoder)">
<meta property="og:url" content="https://yarinamomo.github.io/2022/10/18/DCA/index.html">
<meta property="og:site_name" content="Something about Learning">
<meta property="og:description" content="Recently, machine learning techniques become more and more popular in the field of bioinformatics, especially for analyzing DNA or RNA sequences. The ability of obtaining single-cell RNA sequence (scR">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://github.com/yarinamomo/yarinamomo.github.io/raw/main/images/DCA/autoencoder.png">
<meta property="og:image" content="https://github.com/yarinamomo/yarinamomo.github.io/raw/main/images/DCA/noisemodels.png">
<meta property="og:image" content="https://github.com/yarinamomo/yarinamomo.github.io/raw/main/images/DCA/architecture1.png">
<meta property="og:image" content="https://github.com/yarinamomo/yarinamomo.github.io/raw/main/images/DCA/res1.png">
<meta property="og:image" content="https://github.com/yarinamomo/yarinamomo.github.io/raw/main/images/DCA/res2.png">
<meta property="og:image" content="https://github.com/yarinamomo/yarinamomo.github.io/raw/main/images/DCA/scalability.png">
<meta property="article:published_time" content="2022-10-18T11:06:31.448Z">
<meta property="article:modified_time" content="2022-10-18T12:24:28.327Z">
<meta property="article:author" content="Yiran Wang">
<meta property="article:tag" content="Machine learning">
<meta property="article:tag" content="Bioinformatics">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://github.com/yarinamomo/yarinamomo.github.io/raw/main/images/DCA/autoencoder.png">
  
    <link rel="alternate" href="/atom.xml" title="Something about Learning" type="application/atom+xml">
  
  
    <link rel="shortcut icon" href="/favicon.png">
  
  
    
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/typeface-source-code-pro@0.0.71/index.min.css">

  
  
<link rel="stylesheet" href="/css/style.css">

  
    
<link rel="stylesheet" href="/fancybox/jquery.fancybox.min.css">

  
<meta name="generator" content="Hexo 6.3.0"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">Something about Learning</a>
      </h1>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
          <a class="main-nav-link" href="/about">About</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="Search"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="https://yarinamomo.github.io"></form>
      </div>
    </div>
  </div>
</header>

      <div class="outer">
        <section id="main"><article id="post-DCA" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2022/10/18/DCA/" class="article-date">
  <time class="dt-published" datetime="2022-10-18T11:06:31.448Z" itemprop="datePublished">2022-10-18</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 class="p-name article-title" itemprop="headline name">
      Explanation on DCA (Deep Count Autoencoder)
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p>Recently, machine learning techniques become more and more popular in the field of bioinformatics, especially for analyzing DNA or RNA sequences. The ability of obtaining single-cell RNA sequence (scRNA-seq) opened a larger window for downstream analysis with the high resolution of single cell data compared to the bulk data.</p>
<p>Different measuring techniques for scRNA-seq datasets, however, generate noise which may affect further analysis. The noise consists of amplification bias, library size differences, low RNA capture rate, etc. For example, for the general low RNA capture rate, it may lead to the sequence data with <strong>false</strong> zero counts of certain expressed genes. This kind of values can be seen as <em>missing values</em>, and need to be separated from the <strong>true</strong> zero counts which actually represent the lack of gene expression in the cells. Ideally, the <strong>false</strong> zero counts should be identified and imputed to reconstruct the non-noise version of the scRNA-seq datasets. </p>
<span id="more"></span>

<h3 id="1-DCA"><a href="#1-DCA" class="headerlink" title="1. DCA"></a>1. DCA</h3><p>The paper <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-018-07931-2">Single-cell RNA-seq denoising using a deep count autoencoder</a> by Gökcen Eraslan et al. purposed a machine learning method, namely deep count autoencoder (DCA) to learn a compressed latent representation and to denoise the input scRNA-seq. The method is able to take the non-linearity and the sparsity of the data and scalability of the model into account.</p>
<p>The purposed DCA model is an autoencoder which is a type of neural network which consists a encoder and decoder. The model learns an efficient compression of the input data with a significantly reduced dimension. Thus the compression will lead the model to learn only the representative part of the data and ignore the non-essential variation (noise) so that the model can achieve the goal of denoise. </p>
<p>The following is an example of a standard autoencoder:</p>
<p align="center"> <img width="80%" src="https://github.com/yarinamomo/yarinamomo.github.io/raw/main/images/DCA/autoencoder.png"> </p>
<p align="center"> An example model of a standard autoencoder </p>

<p>For the purposed DCA model, the main idea is that the loss function should be a representation of the <em>noise model</em>. That is to say that the DCA model should minimize the negative likelihood of the distribution of the <em>noise model</em> instead of trying to minimize the difference between the output and the original data itself. According to the sparse structure of the scRNA-seq data, the noise model was suggested with the negative binomial distribution with (ZINB) and without zero-inflation (NB):</p>
<p align="center"> <img width="40%" src="https://github.com/yarinamomo/yarinamomo.github.io/raw/main/images/DCA/noisemodels.png"> </p>
<p align="center"> Formulas of NB and ZINB. where μ and θ are mean and dispersion parameters of the negative binomial component and π is the mixture coefficient that represents the weight of the point mass </p>

<p>To decide on which of the two noise models to use, one can calculate the likelihood ratio on the data (NB&#x2F;ZINB) to see which one gives the higher likelihood. The higher one fits the data better.</p>
<h3 id="2-Evaluation"><a href="#2-Evaluation" class="headerlink" title="2. Evaluation"></a>2. Evaluation</h3><p>To evaluate the DCA model, the authors constructed two experiments. One with simulated data and one with real data. Only the first one is demonstrated here since it is more focus on the method itself.</p>
<p>The simulated data consists of two datasets:</p>
<ul>
<li>dataset1: 2000 cells with 200 genes and two cell types (63% of data values were set to zero)</li>
<li>dataset2: 2000 cells with 200 genes and six cell types (35% of data values were set to zero)</li>
</ul>
<p>The noise model which is the likelihood loss function of the DCA model is ZINB. The demonstration of the DCA model can be seen as below:</p>
<p align="center"> <img width="80%" src="https://github.com/yarinamomo/yarinamomo.github.io/raw/main/images/DCA/architecture1.png"> </p>
<p align="center"> A demonstration of DCA model with ZINB likelihood loss function. (number of output nodes = number of input nodes/genes) </p>

<p>Compared with using the MSE loss function (which tried to reconstruct the input data itself), the result is as following:</p>
<p align="center"> <img width="80%" src="https://github.com/yarinamomo/yarinamomo.github.io/raw/main/images/DCA/res1.png"> </p>
<p align="center"> Result for dataset1 </p>

<p>From left to right, the pictures represent: simulated dataset1 without noise(ground truth), simulated dataset1 with noise(a number of false zeros), denoised with ZINB, denoised with MSE respectively.</p>
<p align="center"> <img width="80%" src="https://github.com/yarinamomo/yarinamomo.github.io/raw/main/images/DCA/res2.png"> </p>
<p align="center"> Result for dataset2 </p>

<p>From left to right, the pictures represent: simulated dataset2 without noise(ground truth), simulated dataset2 with noise(a number of false zeros), denoised with ZINB, denoised with MSE respectively.</p>
<p>From the results, it can be seen clearly that the DCA model with ZINB is able to denoise the data to make the data be able to be separated into classes, and it is much better than the MSE model.</p>
<p>The second experiment was done on a real dataset with 68,579 peripheral blood mononuclear cells and 1,000 highly variable genes (92% zeros). The DCA model was using NB noise model and the latent layer of the DCA model consists of 2 nodes (two dimensional) and it was also proven to be effective. There are also examples of downstream analysis in the paper, for example, clustering, time course modeling, differential expression, protein-RNA co-expression and pseudotime analyses. The detailed result and related analysis are omitted since we are mainly focused on the machine learning method here. For anyone who is interested, please read the original paper ().</p>
<h3 id="3-Scalability"><a href="#3-Scalability" class="headerlink" title="3. Scalability"></a>3. Scalability</h3><p>The DCA method is also scaled linearly with the number of cells. The following is a runtime comparison between DCA and other denoise methods:</p>
<p align="center"> <img width="50%" src="https://github.com/yarinamomo/yarinamomo.github.io/raw/main/images/DCA/scalability.png"> </p>
<p align="center"> Scalability test result </p>

<h3 id="4-Apply"><a href="#4-Apply" class="headerlink" title="4. Apply"></a>4. Apply</h3><p>To apply the method on self data, it is also important to know that the hyperparameters of the DCA model need to be validated based on the dataset at hand, for example, the number of layers, the number of nodes on each layer, optimization function, learning rate, etc. When the data size is limited, it is especially important to use some regularization methods such as dropout layers, L1 or L2 regularization in the DCA neural network.</p>
<p>The DCA model is publicly available at [github](<a target="_blank" rel="noopener" href="https://github.com/">https://github.com/</a> theislab&#x2F;dca). The method is also available in the [preprocessing package] (<a target="_blank" rel="noopener" href="https://scanpy.readthedocs.io/en/latest/generated/scanpy.external.pp.dca.html#">https://scanpy.readthedocs.io/en/latest/generated/scanpy.external.pp.dca.html#</a>) in Scanpy.</p>

      
    </div>
    <footer class="article-footer">
      <a data-url="https://yarinamomo.github.io/2022/10/18/DCA/" data-id="cl9e6copk0002fpcq14og6zvl" data-title="Explanation on DCA (Deep Count Autoencoder)" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Bioinformatics/" rel="tag">Bioinformatics</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Machine-learning/" rel="tag">Machine learning</a></li></ul>

    </footer>
  </div>
  
    
<nav id="article-nav">
  
    <a href="/2022/11/11/TPE/" id="article-nav-newer" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Newer</strong>
      <div class="article-nav-title">
        
          Hyperparameter Optimization TPE
        
      </div>
    </a>
  
  
    <a href="/2022/10/12/GPUCPU/" id="article-nav-older" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">Is GPU much faster than CPU in machine learning?</div>
    </a>
  
</nav>

  
</article>




  <div id="disqus_thread"></div>
  <script>
      /**
      *  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
      *  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables    */
      var disqus_config = function () {
        this.page.url = 'https://yarinamomo.github.io/2022/10/18/DCA/';  // Replace PAGE_URL with your page's canonical URL variable
        this.page.identifier = '1666091191448'; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
      };
      (function() { // DON'T EDIT BELOW THIS LINE
      var d = document, s = d.createElement('script');
      s.src = 'https://https-yarinamomo-github-io.disqus.com/embed.js';
      s.setAttribute('data-timestamp', +new Date());
      (d.head || d.body).appendChild(s);
      })();
  </script>
  <noscript>Please enable JavaScript to view the <a target="_blank" rel="noopener" href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</section>
        
          <aside id="sidebar">
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">ABOUT</h3>
    <div class="widget">
      <div class="row">
        <img src="https://github.com/yarinamomo/yarinamomo.github.io/raw/main/images/profile.jpg" alt="profile picture" width=80% >
      </div>
      Email: <b>yarinawangmomo@gmail.com</b>
      LinkedIn：<a target="_blank" rel="noopener" href="https://www.linkedin.com/in/yiran-wang-a6201a9a/">Yiran Wang</a>
    </div>
  </div>


  
    

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Tags</h3>
    <div class="widget">
      <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bioinformatics/" rel="tag">Bioinformatics</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Computer-Vision/" rel="tag">Computer Vision</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Configuration/" rel="tag">Configuration</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Design-Pattern/" rel="tag">Design Pattern</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Machine-learning/" rel="tag">Machine learning</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Programming/" rel="tag">Programming</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Static-Analysis/" rel="tag">Static Analysis</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Tag Cloud</h3>
    <div class="widget tagcloud">
      <a href="/tags/Bioinformatics/" style="font-size: 10px;">Bioinformatics</a> <a href="/tags/Computer-Vision/" style="font-size: 10px;">Computer Vision</a> <a href="/tags/Configuration/" style="font-size: 13.33px;">Configuration</a> <a href="/tags/Design-Pattern/" style="font-size: 10px;">Design Pattern</a> <a href="/tags/Machine-learning/" style="font-size: 16.67px;">Machine learning</a> <a href="/tags/Programming/" style="font-size: 20px;">Programming</a> <a href="/tags/Static-Analysis/" style="font-size: 10px;">Static Analysis</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Recent Posts</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2023/02/01/sa1/">Static Program Analysis (1) - Intro and IR</a>
          </li>
        
          <li>
            <a href="/2023/01/07/designpatterns/">Software Design Pattern (1) - Creational, Factory</a>
          </li>
        
          <li>
            <a href="/2022/12/09/codeadvent9/">Christmas Advent of Code 2022 - Day 9</a>
          </li>
        
          <li>
            <a href="/2022/12/05/DataStructure/">Exploring some Data Structures</a>
          </li>
        
          <li>
            <a href="/2022/11/22/CSharp/">Interesting things in C#</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      
      &copy; 2023 Yiran Wang<br>
      Powered by <a href="https://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>

    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
</nav>
    


<script src="/js/jquery-3.4.1.min.js"></script>



  
<script src="/fancybox/jquery.fancybox.min.js"></script>




<script src="/js/script.js"></script>





  </div>
</body>
</html>